File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(int n, long long left, long long right) {
    vector<int> answer;
    
    for(long long i=left; i<=right; i++){
        long long div = i/n;
        long long mod = i%n;
        
        long long bigger = max(div, mod);
        
        answer.push_back(bigger+1);
    }
    
    return answer;
}

File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.js
function solution(n, left, right) {
    var answer = [];
    
    for(let i=left; i<=right; i++){
        let div = Math.floor(i/n);
        let mod = i%n;
        
        answer.push(Math.max(div, mod)+1);
    }
    
    return answer;
}